BootloaderCommonPkg/Library/PagingLib/PagingMap.c

```diff
@@ -47,13 +106,20 @@ MapMemoryRange (
   UINTN          EntryNum;
   UINTN         *PageTable;
   UINTN         *PageTable4K;
-  UINT32         PageBase;
+  UINTN          PageBase;
   UINT32         Attribute;
-  UINT32         Address;
+  UINTN          Address;
   UINT32         PteOffset;
   UINT32         PdeOffset;
-  UINT32         Alignment;
+  UINTN          Alignment;
   UINT32         PageBits;
+  EFI_STATUS     Status;
+
+  if (Ranges[0].Start >= 0x180000000ULL) {

```

Comment:
Why do we do this ?
I think we can always check to see if the entry is valid for a given address.  If it is not there, then we can try to add entry.